### ISSUE TITLE
Filenames on server that have a colon in them (:) appear to cause a recursion loop in mkdirp

### ISSUE THREAD

**Issue by justinross** [ThatOdieGuy/downspout#32]
```
{"level":"info","message":"mkdirp j:/destFolder/Like this: A folder name that has a colon in it/","timestamp":"2017-08-25T22:28:34.809Z"}
{"date":"Fri Aug 25 2017 15:28:35 GMT-0700 (Pacific Daylight Time)","process":{"pid":5848,"uid":null,"gid":null,"cwd":"C:\\Users\\username\\downspout\\htpc","execPath":"C:\\Program Files\\nodejs\\node.exe","version":"v6.11.1","argv":["C:\\Program Files\\nodejs\\node.exe","C:\\Users\\username\\downspout\\htpc\\server\\build\\server\\src\\bin\\www.js"],"memoryUsage":{"rss":245243904,"heapTotal":218169344,"heapUsed":161868008,"external":3373106}},"os":{"loadavg":[0,0,0],"uptime":54264.1710375},"trace":[{"column":18,"file":"fs.js","function":"Object.fs.mkdirSync","line":923,"method":"mkdirSync","native":false},{"column":13,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":71,"method":null,"native":false},{"column":24,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":77,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"username":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false},{"column":17,"file":"C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js","function":"sync","line":78,"method":null,"native":false}],"stack":["RangeError: Maximum call stack size exceeded","    at Object.fs.mkdirSync (fs.js:923:18)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:71:13)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:77:24)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)","    at sync (C:\\Users\\username\\downspout\\htpc\\node_modules\\mkdirp\\index.js:78:17)"],"level":"error","message":"uncaughtException: Maximum call stack size exceeded","timestamp":"2017-08-25T22:28:35.696Z"}

```

**Comment by ThatOdieGuy** [comment]
Fixed with 12f4f4b19f6f96e20345e6e419ba855f4774b15f

Can you please stop downloading movies with weird file names in them? 💔

**Comment by justinross** [comment]
This is top-notch beta testing!
